x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/request/elastic/ElasticInferenceServiceRequest.java

```diff
@@ -7,6 +7,30 @@
 
 package org.elasticsearch.xpack.inference.external.request.elastic;
 
+import org.apache.http.client.methods.HttpRequestBase;
+import org.elasticsearch.tasks.Task;
+import org.elasticsearch.xpack.inference.external.request.HttpRequest;
 import org.elasticsearch.xpack.inference.external.request.Request;
 
-public interface ElasticInferenceServiceRequest extends Request {}
+public abstract class ElasticInferenceServiceRequest implements Request {
+
+    private final String productOrigin;
+
+    public ElasticInferenceServiceRequest(String productOrigin) {
+        this.productOrigin = productOrigin;
+    }
+
+    public String getProductOrigin() {
+        return productOrigin;
+    }
+
+    @Override
+    public final HttpRequest createHttpRequest() {
+        HttpRequestBase request = createHttpRequestBase();
+        // TODO: consider moving tracing here, too
```

Comment:
Wanted to keep it out of scope for this PR, but I think it also makes sense to move the tracing logic here, so we don't risk to forget it for new request subclasses